src/constants.integrations.ts

```diff
@@ -7,6 +7,7 @@ export enum HostingIntegrationId {
 
 export enum SelfHostedIntegrationId {
 	GitHubEnterprise = 'github-enterprise',
+	CloudGitHubEnterprise = 'cloud-github-enterprise',
```

Comment:
I'm definitely very late to this conversation but, would you mind clarifying why this needs to be a different key and not just the same "github-enterprise" we use for the non-cloud one (your connection is either one or the other, right? but both are GH enterprise, just like how we treat regular github integrations).